tests/python/gaia-ui-tests/gaiatest/apps/marketplace/app.py

```diff
@@ -70,3 +72,57 @@ def tap_install_button(self):
 
     def get_app_name(self):
         return self.root_element.find_element(*self._name_locator).text
+
+class MarketplaceDev(Marketplace):
+
+    name = 'Dev'
+    manifest_url = 'https://marketplace.firefox.com/app/bee10fbb-7829-4d99-8a9f-a6d0a905138e/manifest.webapp'
+
+    _enable_addons_locator = (By.ID, 'enable-addons')
+
+    def _enable_addons(self):
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._enable_addons_locator))
+        Wait(self.marionette).until(expected.element_displayed(element))
+        element.tap()
+
+    def _open_addon(self, addon_name):
+        selector = '//a[text()="%s"]' % addon_name
+        addon_locator = (By.XPATH, selector)
+
+        element = Wait(self.marionette, timeout=20, interval=1).until(
+            expected.element_present(*addon_locator))
+        Wait(self.marionette).until(expected.element_displayed(element))
+        element.tap()
+        return AddonPage(self.marionette)
+
+    def install_addon(self, addon_name):
+        self.search(':debug')
+
+        # Debug page is opened, tap on the "Enable Firefox OS Add-ons" button
+        self._enable_addons()
+
+        # Addons page gets loaded, tap on the addon link to open that page
+        addon_page = self._open_addon(addon_name)
+        addon_page.install()
+
+class AddonPage(Base):
+
+    _addon_install_locator = (By.CSS_SELECTOR, '.addon-install-btn')
+
+    def install(self):
+        element = Wait(self.marionette, timeout=20, interval=1).until(
+            expected.element_present(*self._addon_install_locator))
+        Wait(self.marionette).until(expected.element_displayed(element))
+        element.tap()
+
+        # Confirm the installation and wait for the app icon to be present
```

Comment:
Yes, I copied this code from somewhere, probably from one of the homescreen tests.